plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -634,11 +634,46 @@ protected JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorT
         }
     }
 
-    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    /**
+     * Allow some connectors that do not support directly setting comment when creating the table to return a list of additional comment setting statements here,
+     * the returned statements will be executed with the `create table` statement in the same transaction.
+     */
+    protected List<String> getExtraTableAndColumnComments(RemoteTableName remoteTableName, ConnectorSession session, Connection connection, ConnectorTableMetadata tableMetadata)
     {
         if (tableMetadata.getComment().isPresent()) {
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with table comment");
         }
+        return ImmutableList.of();
+    }
+
+    protected void execute(ConnectorSession session, Connection connection, String query, List<String> extraComments)
```

Comment:
Change `execute(ConnectorSession session, Connection connection, String query)` to accept `execute(ConnectorSession session, Connection connection, String... queries)` and please do not commit there.

Reply:
Do you mean execute the sqls directly without `setAutoCommit to false`?

Reply:
If so, I would prefer adding a new method, accept the `List<String> sqls`, and iterator the `sqls`, call the original `execute` method

Reply:
The question is how many transactions we will have with such approach. I would prefer single one if possible.

Reply:
👍 Got it